ZZ_testing_channels.md

```diff
@@ -0,0 +1,163 @@
+Testing Phoenix Channels will help ensure that your code continues to work
+as you make changes to your application. It helps guide development and
+ensure there is no breakage later on down the road.
+
+Phoenix has built in helper functions in `Phoenix.Channel.Test` that help
+make testing Channels easy.
+
+### Testing `join` authorization
+
+Let’s add a new file at `test/channels/room_channel_test.exs`
+
+    defmodule App.RoomChannelTest do
+      use ExUnit.Case
+      import Phoenix.Channel.Test
+      alias App.RoomChannel
+
+      test "room:join requires the super secure auth token" do
+        {status, _socket} =
+          build_socket("room:join")
+          |> join(RoomChannel, %{"token" => "phoenix"}
+
+        assert status == :ok
+      end
+    end
+
+So what’s happening here?
+
+First we’ve set this test up with `ExUnit` and imported the test helpers
+with `import Phoenix.Channel.Test`.
+
+Next, we build a socket with the topic `"room:join"` and join the
+`RoomChannel` with the params `%{"token" => "phoenix"}`. We pattern match to
+get the status of the response and check that it is `:ok` meaning that the
+socket is authorized.
+
+When we run `mix test` we see that we have not yet defined
+`App.RoomChannel`. Let’s get this test passing by creating a file at
+`web/channels/room_channel.ex`
```

Comment:
I definitely like both the explanation and the progression here, i.e. nice TDD form. :)